Tidy up files

gcapes opened the pull request:
An initial stab at #1225.

gcapes commented:
This will want checking for errors, but also I'd like to discuss the best solution to some of the extra complexity the data and scripts directory has introduced.

Episode 5 has only a small modification to access the `goostats.sh` script, which is now `../scripts/goostats.sh`.
Episode 6 starts to get complicated, which is probably why there hasn't been a separation of data and scripts yet.

If we were to be doing this IRL, we'd add the scripts to PATH so they could be easily accessed, given the script we write calls the other we already have (`goostat.sh`). PATH isn't covered so we can't do that. The problem then is that one of the scripts makes a copy of the original data files `stats-$datafile` which means it has to be run from the data directory to avoid polluting the scripts directory with back ups of data files. There is an error I just spotted where the `do-stats.sh` script calls `bash goostats.sh` but it should be `bash ../scripts/goostats.sh`. But anyway, this hard-coding of the script file path is jarring and undermines the flexibility that is taught by using `$@` to run on any user input.

So, put data and scripts back into the same file? Yes it's an anti-pattern, but without covering PATH I think it might be the best solution. Having the scripts just one level up from the `data` directory doesn't solve the relative path issue. It's going to have to be a compromise somewhere, the question is where?

deppen8 commented:
@gcapes, I think I was the one who raised the "scripts next to data" as a bit "smelly", but I want to stress that I don't feel too strongly about that. If scripts next to data makes the whole thing run with minimal disruption, we should do that.

deppen8 commented:
@gcapes , is this ready for detailed review yet?

gcapes commented:
Yep, please go ahead!

deppen8 commented on _episodes/06-script.md:
```diff
@@ -550,7 +558,7 @@ Of course, this introduces another tradeoff between flexibility and complexity.
 > ## Script Reading Comprehension
 >
-> For this question, consider the `shell-lesson-data/molecules` directory once again.
+> For this question, consider the `shell-lesson-data/exercise-data/proteins` directory once again.
```
This one is OK, but there are still a number of references to `molecules` in this lesson. Probably just need to find/replace those.

deppen8 reviewed (changes requested):
Thanks for all the hard work you have put in here, @gcapes. To review, I did a fresh `make serve` of the lesson from the PR then went through the whole thing as an instructor/learner would: download the data, start from episode 1, and carry on through episode 7. There are a few rough edges to smooth out, but hopefully those are not too onerous.

I made individual comments/suggestions where I could on the diffs. There were some places where I couldn't put comments directly in the diffs because they were on parts that you didn't change (yet). Those are:

- The path that leads off episode 7 needs to be `Desktop/shell-lesson-data/exercise-data/writing`
- The path in the Little Women exercise at the end of episode 7 also needs to be updated: `shell-lesson-data/writing/exercise-data/writing/LittleWomen.txt`

gcapes commented on _episodes/02-filedir.md:
```diff
@@ -751,13 +749,11 @@ For example, `ls -s` will display the size of files and directories alongside th
-total 116
- 4 amino-acids.txt   4 animals.txt   4 morse.txt  12 planets.txt  76 sunspot.txt
- 4 animal-counts     4 elements      4 pdb         4 salmon.txt
+$ ls -s Desktop/shell-lesson-data/exercise-data/
+total 28
+ 4 animal-counts   4 creatures  12 numbers.txt   4 proteins   4 writing
```
This must be an OS thing - I've just double checked the values. Which OS do you use?

gcapes commented on _episodes/02-filedir.md:
```diff
@@ -751,13 +749,11 @@ For example, `ls -s` will display the size of files and directories alongside th
 $ ls -S Desktop/shell-lesson-data/data
-sunspot.txt  animal-counts  pdb        amino-acids.txt  salmon.txt
-planets.txt  elements       morse.txt  animals.txt
+animal-counts  creatures  proteins  writing  numbers.txt
```
I wonder what is going on here. This is what my bash is reporting...
```
$ du . | sort -n
4	./writing/thesis
16	./animal-counts
40	./creatures
76	./proteins
1060	./writing
1208	.
```

gcapes commented:
Thanks for the review!  It's a lot to work through. I think I've addressed all your comments, but there's also some stuff in episode 7 I've overlooked in that the filesystem diagram and subsequent output is now incorrect. I'll add a couple of to-do items below so I don't forget.

- [ ]  Update diagram to match current file structure
- [ ] Update subsequent output

gcapes commented:
The only thing which is (possibly) outstanding is an updated file tree diagram, but I think it would be too large given the new directory layout.
Is this ok to merge now if I fix the conflicts? @swcarpentry/shell-novice-maintainers

deppen8 commented:
I say merge. The only way we can know the rough edges is if more people start using it.

gcapes commented:
I've just replaced the `ls` output with the visually easier output from `tree`. I don't think it matters that we've not covered `tree`, and I've not shown the command - just the output. If no-one objects (?) I'll go ahead and merge this, and people can start teaching it.

deppen8 commented:
Just want to add a comment to commemorate this PR 😄 getting merged. It was a huge lift by @gcapes and I am really excited to see how it improves the lesson. 🥳
